### PR TITLE
PyDCP simple_dcp_client feature additions

### PR DESCRIPTION
**Pull request by CSubmerged**
Details are contained within the commits but as an overview:
- Re-formatted files to PEP 8 standards

- Added new arguments:
-> Timeout - Sets DCPClient timeout length, also allows persistence
-> NOOP interval - Sets length between DCPClient NOOP requests
-> Opcode dump - prints opcodes and names received by the client 

- Implemented NOOP acknowledgement

### CODE REVIEW

File: simple_dcp_client.py [hunk 1]
```diff
@@ -200,6 +203,7 @@ def parseArguments():
     parser.add_argument("--delete_times", help="Include delete times", default=False, required=False,
                         action="store_true")
     parser.add_argument("--compression", '-y', help="Compression", required=False, action='count', default=0)
+    parser.add_argument("--timeout", '-t', help="Set timeout length, -1 forces persistence", required=False, default=5)
```
Comment (dhaikney) [inline simple_dcp_client.py]: needs to specify the units (seconds). persistence is accurate but can be ambiguous. "Disables timeout" might be clearer

File: simple_dcp_client.py [hunk 2]
```diff
@@ -163,6 +164,10 @@ def initiate_connection(args):
     assert response['status'] == SUCCESS
     print "Enabled NOOP"
 
+    response2 = dcp_client.general_control("set_noop_interval", noop_interval)
```
Comment (dhaikney) [inline simple_dcp_client.py]: Most DCP clients wont specify this so we don't want to do it all the time (it's quite a rare option). As such, I would make this conditional on the argument being specified. No need to issues the control command to reinstate the default. e.g. : if args.noop_interval:

File: lib/dcp_bin_client.py [hunk 3]
```diff
@@ -242,6 +249,28 @@ def ack_dcp_noop_req(self, opaque):
                              0, 0, 0, 0, 0, opaque, 0)
         self.s.sendall(header)
 
+    def opcode_dump_control(self, control):
+        self.__opcode_dump = control
+        
+    def opcode_lookup(self, opcode):
```
Comment (dhaikney) [inline lib/dcp_bin_client.py]: This is quite heavy duplication of the constants file. If we were ever to change (or add a new) opcode we'd only want to do this in one place. I'd suggest moving the hex-string->CMDstring lookup to the constants file and possibly doing it based on the int value so you can do constant to string instead CMD_MUTATION->"CMD_MUTATION"

**Comment by CSubmerged** [comment]
Change requests have been addressed. Please check that they satisfy your requirements. Thanks

**Pull request merged**